src/client/api/events/sub_outputs_payload.c

```diff
@@ -4,108 +4,122 @@
-      printf("[%s:%d]: gets %s json uint32 failed\n", __func__, __LINE__, JSON_KEY_TYPE);
-      goto end;
-    }
-
-    // amount
-    if ((ret = json_get_uint64(j_output, JSON_KEY_AMOUNT, &res->output.amount)) != 0) {
-      printf("[%s:%d]: gets %s json uint64 failed\n", __func__, __LINE__, JSON_KEY_AMOUNT);
-      goto end;
-    }
-
-    // address object
-    cJSON *j_addr = cJSON_GetObjectItemCaseSensitive(j_output, JSON_KEY_ADDR);
-    if (j_addr) {
-      // ed25519 address
-      if ((ret = json_get_string(j_addr, JSON_KEY_ADDR, res->output.addr, sizeof(res->output.addr))) != 0) {
-        printf("[%s:%d]: parsing %s failed\n", __func__, __LINE__, JSON_KEY_ADDR);
-        goto end;
-      }
-    } else {
-      printf("[%s:%d]: address object not found\n", __func__, __LINE__);
-      ret = -1;
-      goto end;
-    }
-  } else {
-    printf("[%s:%d]: output object not found\n", __func__, __LINE__);
-    ret = -1;
-    goto end;
+  iota_str_t *topic_buff = NULL;
+  char const *const topic_str = "outputs/unlock/";
+  // outputs/unlock/{condition}/{address}/spent
+  topic_buff =
+      iota_str_reserve(strlen(topic_str) + strlen(unlock_condition) + BIN_TO_HEX_STR_BYTES(ADDRESS_MAX_BYTES) + 8);
+  if (topic_buff == NULL) {
+    printf("[%s:%d]: allocate command buffer failed\n", __func__, __LINE__);
+    return -1;
   }
 
-end:
+  // composing API command
+  snprintf(topic_buff->buf, topic_buff->cap, "%s%s%s%s%s", topic_str, unlock_condition, "/", addr_bech32, "/spent");
+  topic_buff->len = strlen(topic_buff->buf);
 
-  cJSON_Delete(json_obj);
+  int ret = event_subscribe(client, mid, topic_buff->buf, qos);
+  iota_str_destroy(topic_buff);
   return ret;
 }
 
-int event_sub_address_outputs(event_client_handle_t client, int *mid, char const addr[], bool is_bech32, int qos) {
-  if ((strlen(addr)) != IOTA_ADDRESS_HEX_BYTES) {
-    printf("[%s:%d]: Address length is invalid\n", __func__, __LINE__);
+int event_sub_outputs_alias_id(event_client_handle_t client, int *mid, char const alias_id[], int qos) {
+  if (strlen(alias_id) != BIN_TO_HEX_BYTES(ALIAS_ID_BYTES)) {
+    printf("[%s:%d] incorrect length of id\n", __func__, __LINE__);
     return -1;
   }
-  // 91 is the max length of string addresses/ed25519/[addr_str]/outputs
-  char topic_buff[91] = {};
-  if (is_bech32) {
-    sprintf(topic_buff, "addresses/%s/outputs", addr);
-  } else {
-    sprintf(topic_buff, "addresses/ed25519/%s/outputs", addr);
+
+  // Buffer enough for outputs/aliases/{aliasId}
+  char topic_buff[59] = {};
+  sprintf(topic_buff, "outputs/aliases/0x%s", alias_id);
+
+  return event_subscribe(client, mid, topic_buff, qos);
+}
+
+int event_sub_outputs_nft_id(event_client_handle_t client, int *mid, char const nft_id[], int qos) {
+  if (strlen(nft_id) != BIN_TO_HEX_BYTES(NFT_ID_BYTES)) {
+    printf("[%s:%d] incorrect length of id\n", __func__, __LINE__);
+    return -1;
   }
+
+  // Buffer enough for outputs/nfts/{nftId}
+  char topic_buff[56] = {};
```

Comment:
Can this number be somehow calculated or maybe create additional constant for it?